src/data/finset/basic.lean

```diff
@@ -2590,6 +2607,16 @@ theorem product_eq_bUnion [decidable_eq α] [decidable_eq β] (s : finset α) (t
 ext $ λ ⟨x, y⟩, by simp only [mem_product, mem_bUnion, mem_image, exists_prop, prod.mk.inj_iff,
   and.left_comm, exists_and_distrib_left, exists_eq_right, exists_eq_left]
 
+@[simp] lemma product_bUnion {β γ : Type*} [decidable_eq γ] (A : finset α) (B : finset β)
+  (f : α × β → finset γ) :
+  (A.product B).bUnion f = A.bUnion (λ a, B.bUnion (λ b, f (a, b))) :=
+begin
+  ext x,
+  simp only [mem_bUnion, exists_prop, mem_product],
+  exact ⟨λ ⟨⟨a, b⟩, ⟨ha, hb⟩, hx⟩, ⟨a, ha, b, hb, hx⟩,
+    λ ⟨a, ha, b, hb, hx⟩, ⟨⟨a, b⟩, ⟨ha, hb⟩, hx⟩⟩,
+end
```

Comment:
Does this fol;ow from `product_eq_bUnion` + some other lemmas?

Reply:
No, because it doesn't require `decidable_eq α`

Reply:
I mean that doesn't really matter, you can just use `classical`. Does the current proof avoid choice according to `#print axioms`?

Reply:
Here's the proof I'm thinking of:
```lean
@[simp] lemma product_bUnion {β γ : Type*} [decidable_eq γ] (A : finset α) (B : finset β)
  (f : α × β → finset γ) :
  (A.product B).bUnion f = A.bUnion (λ a, B.bUnion (λ b, f (a, b))) :=
begin
  classical,
  simp_rw [product_eq_bUnion, bUnion_bUnion, image_bUnion],
end
```
which reveals we're missing this lemma (which I didn't attempt to golf):
```
lemma bUnion_bUnion [decidable_eq γ] (s : finset α) (f : α → finset β) (g : β → finset γ) :
  (s.bUnion f).bUnion g = s.bUnion (λ a, (f a).bUnion g) :=
begin
  ext,
  simp only [finset.mem_bUnion, exists_prop],
  simp_rw [←exists_and_distrib_right, ←exists_and_distrib_left, and_assoc],
  rw exists_comm,
end
```

Reply:
It yields
```
propext
quot.sound
classical.choice
```
so i guess it does use it? Your proof gives the same list of axioms.

Reply:
I didn't manage to golf your proof of `bUnion_bUnion`.